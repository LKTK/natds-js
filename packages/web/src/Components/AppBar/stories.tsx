import * as React from "react";
import { makeStyles } from "@material-ui/core/styles";
import AppBar from ".";
import Toolbar from "../Toolbar";
import IconButton from "../IconButton";
import Typography from "../Typography";
import Button from "../Button";
import Icon from "../Icon";
import {IStorySections} from "../types/IStorySections";

interface IAppBarWrapper {
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   color: any;
   elevation?: number;
}

const AppBarWrapper = (props: IAppBarWrapper) => {
  const { color, elevation: elevationProp } = props;

  const useStyles = makeStyles((theme) => ({
    root: {
      flexGrow: 1,
      // eslint-disable-next-line no-magic-numbers
      width: theme.spacing(62),
    },
    menuButton: {
      // eslint-disable-next-line no-magic-numbers
      marginRight: theme.spacing(2),
    },
    title: {
      flexGrow: 1,
    },
  }));

  const classes = useStyles();

  return (
    <div className={classes.root}>
      <AppBar position="static" color={color} elevation={elevationProp}>
        <Toolbar>
          <IconButton edge="start" className={classes.menuButton} color="inherit" aria-label="menu">
            <Icon name="filled-navigation-menu" size="tiny" />
          </IconButton>
          <Typography variant="h6" className={classes.title}>
            AppBar
          </Typography>
          <Button color="inherit">Login</Button>
        </Toolbar>
      </AppBar>
    </div>
  );
};

export const colors : IStorySections = [
  {
    component: <AppBarWrapper color="default" />,
    title: "default",
  },
  {
    component: <AppBarWrapper color="inherit" />,
    title: "inherit",
  },
  {
    component: <AppBarWrapper color="primary" />,
    title: "primary",
  },
  {
    component: <AppBarWrapper color="secondary" />,
    title: "secondary",
  },
];

export const elevation = [
  {
    component: <AppBarWrapper color="primary" />,
    title: "Default elevation",
  },
  {
    component: <AppBarWrapper color="primary" elevation={0} />,
    title: "Elevation equals zero",
  },
];
