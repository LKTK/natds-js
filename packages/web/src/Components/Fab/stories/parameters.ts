export const parameters = {
  jestImportPath: "web",
  jest: ["Fab"],
  theme: {
    context: "web",
  },
};
