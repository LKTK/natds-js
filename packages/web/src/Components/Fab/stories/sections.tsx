import * as React from "react";
import { Fab, Icon } from "@naturacosmeticos/natds-web";

export const colors = [
  {
    title: "Primary",
    component: <Fab color="primary"><Icon name={"outlined-action-add"} size={"standard"} /></Fab>,
  },
  {
    title: "Secondary",
    component: <Fab color="secondary"><Icon name={"outlined-action-add"} size={"standard"} /></Fab>,
  },
  {
    title: "Light",
    component: <Fab color="light"><Icon name={"outlined-action-add"} size={"standard"} /></Fab>,
  },
];

export const variants = [
  {
    title: "Round",
    component: <Fab variant="round"><Icon name={"outlined-action-add"} size={"standard"} /></Fab>,
  },
  {
    title: "Extended",
    component: <Fab variant="extended">Extended (with text)</Fab>,
  },
];

export const sizes = [
  {
    title: "Large",
    component: <Fab size="large"><Icon name={"outlined-action-add"} size={"standard"} /></Fab>,
  },
  {
    title: "Medium",
    component: <Fab size="medium"><Icon name={"outlined-action-add"} size={"standard"} /></Fab>,
  },
  {
    title: "Small",
    component: <Fab size="small"><Icon name={"outlined-action-add"} size={"small"} /></Fab>,
  },
];

export const leadingIcon = [
  {
    title: "Leading Icon",
    component: <Fab variant="extended"><Icon name={"outlined-action-add"} size={"small"} /> Icon and text</Fab>,
  },
];

export const disabled = [
  {
    title: "Disabled",
    component: <Fab disabled><Icon name={"outlined-action-add"} size={"standard"} /></Fab>,
  },
];
