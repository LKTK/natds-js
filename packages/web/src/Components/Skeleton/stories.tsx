import * as React from "react";
import { tokens } from "@naturacosmeticos/natds-styles";
import Skeleton from ".";
import {IStorySections} from "../types/IStorySections";
import Avatar from "../Avatar";
import Typography, {TypographyVariant} from "../Typography";

export const variants: IStorySections = [
  {
    component: <Skeleton variant={"text"} />,
    title: "Text",
  },
  {
    component: <Skeleton variant={"circle"} height={tokens.sizes.semix} width={tokens.sizes.semix} />,
    title: "Circle: semix size (40px)",
  },
  {
    component: <Skeleton variant={"rect"} height={tokens.sizes.hugex} width={tokens.sizes.veryhuge} />,
    title: "Rect",
  },
];

export const animations: IStorySections = [
  {
    component: <Skeleton width={tokens.sizes.veryhuge} />,
    title: "Default animation (pulsate)",
  },
  {
    component: <Skeleton animation={false} width={tokens.sizes.veryhuge} />,
    title: "No animation",
  },
  {
    component: <Skeleton animation="wave" width={tokens.sizes.veryhuge} />,
    title: "Wave animation",
  },
];

const typographyVariants : TypographyVariant[] = ["h1",
  "h3",
  "body1",
  "caption"];

export const forTypography: IStorySections = typographyVariants.map((variant) => ({
  component: <Typography variant={variant}><Skeleton /></Typography>,
  title: variant,
}));

const avatarAltText = "Nat Natura Avatar";
const avatarUrl = "https://pbs.twimg.com/profile_images/1285932878795333632/Wl95KE_m_400x400.jpg";

export const forAvatar : IStorySections = [
  {
    component: <Skeleton variant={"circle"}><Avatar alt={avatarAltText} size={"huge"} src={avatarUrl} /></Skeleton>,
    title: "Avatar (loading)",
  },
  {
    component: <Avatar alt={avatarAltText} size={"huge"} src={avatarUrl} />,
    title: "Avatar (loaded)",
  },
];
