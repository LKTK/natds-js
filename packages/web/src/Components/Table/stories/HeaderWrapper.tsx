import * as React from "react";

import Icon from "../../Icon";
import { Table } from "../Table";
import { TableCell } from "../TableCell";
import { TableHead } from "../TableHead";
import { TableRow } from "../TableRow";
import { TableSortLabel } from "../TableSortLabel";

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export const HeaderWrapper = ({
  active,
  direction,
  IconComponent,
}:{
    active?: boolean,
    direction?: "asc" | "desc",
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    IconComponent?: any
  }) => {

  const DefaultIcon = ({componentActive, componentDirection, ...props}: { componentActive?: boolean, componentDirection ?: "asc" | "desc" }) => {
    let iconName = "outlined-action-order";

    if (componentActive && componentDirection === "desc") {
      iconName = "outlined-navigation-directionbottom";
    }

    if (componentActive && componentDirection === "asc") {
      iconName = "outlined-navigation-directiontop";
    }

    return <Icon name={iconName} size="small" {...props} />;
  };

  return (
    <div>
      <Table>
        <TableHead>
          <TableRow>
            <TableCell align="center">
              <TableSortLabel
                active={active}
                onClick={() => { Function.prototype(); }}
                IconComponent={IconComponent || DefaultIcon({componentActive: active, componentDirection: direction})}
              >
                  Cell with Sort Icon
              </TableSortLabel>
            </TableCell>
          </TableRow>
        </TableHead>
      </Table>
    </div>
  );
};
