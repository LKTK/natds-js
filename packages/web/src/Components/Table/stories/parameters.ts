export const parameters = {
  jestImportPath: "web",
    jest: ["Table"],
    theme: {
    context: "web",
  }
};
