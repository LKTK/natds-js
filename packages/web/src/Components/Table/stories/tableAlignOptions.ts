export const tableAlignOptions: { left: string; justify: string; center: string; inherit: string; right: string } = {
  center: "center",
  inherit: "inherit",
  justify: "justify",
  left: "left",
  right: "right",
};
