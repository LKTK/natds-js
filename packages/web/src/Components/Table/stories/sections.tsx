import * as React from "react";

import { TableHeaderCell } from "./TableHeaderCell"
import { TableWrapper } from "./TableWrapper"
import { TableWithSelectWrapper } from "./TableWithSelectWrapper"
import { HeaderWrapper } from "./HeaderWrapper"

export const tableContainerVariant = [
  {
    title: "standard",
    component: <TableWrapper/>,
  },
];

export const tableContainerAttribute = [
  {
    title: "rounded: on",
    component: <TableWrapper rounded/>,
  },
  {
    title: "dividers: none",
    component: <TableWrapper dividers={false}/>,
  },
  {
    title: "striped: none",
    component: <TableWrapper striped={false}/>,
  },
  {
    title: "Table Header Cell / Ordination desc",
    component: <HeaderWrapper active direction={"desc"}/>,
  },
  {
    title: "Table Header Cell / Ordination asc",
    component: <HeaderWrapper active direction={"asc"}/>,
  },
  {
    title: "Table Header Cell / Ordination order",
    component: <HeaderWrapper active/>,
  },
  {
    title: "Table Header Cell / Selection control",
    component: <TableHeaderCell/>,
  },
  {
    title: "Table Header Cell / Selection control",
    component: <TableHeaderCell/>,
  },
  {
    title: "Table Header Cell / Selection control indeterminate",
    component: <TableHeaderCell indeterminate/>,
  },
  {
    title: "Table Data Cell / Align: Left",
    component: <TableWrapper align={"left"} />,
  },
  {
    title: "Table Data Cell / Align: Center",
    component: <TableWrapper align={"center"}/>,
  },
  {
    title: "Table Data Cell / Align: Right",
    component: <TableWrapper align={"right"}/>,
  },
  {
    title: "Row / Selected",
    component: <TableWithSelectWrapper align={"center"}/>,
  },
];
