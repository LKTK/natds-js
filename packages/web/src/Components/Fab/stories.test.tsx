import * as renderer from "react-test-renderer";
import {colors} from "./stories";

describe("Fab", () => {
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  describe.each(colors)("when color is %p", (color) => {
    it("should render correctly", () => {
      const wrapper = renderer.create(color.component);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
