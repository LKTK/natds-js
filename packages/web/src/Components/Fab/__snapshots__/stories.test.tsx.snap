// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fab when color is {"component": [Object], "title": "Light"} should render correctly 1`] = `
<button
  className="MuiButtonBase-root MuiFab-root makeStyles-root-9"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiFab-label"
  >
    <i
      className="natds-icons natds-icons-outlined-action-add makeStyles-root-12"
    />
  </span>
</button>
`;

exports[`Fab when color is {"component": [Object], "title": "Primary"} should render correctly 1`] = `
<button
  className="MuiButtonBase-root MuiFab-root makeStyles-root-1 MuiFab-primary makeStyles-primary-2"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiFab-label"
  >
    <i
      className="natds-icons natds-icons-outlined-action-add makeStyles-root-4"
    />
  </span>
</button>
`;

exports[`Fab when color is {"component": [Object], "title": "Secondary"} should render correctly 1`] = `
<button
  className="MuiButtonBase-root MuiFab-root makeStyles-root-5 MuiFab-secondary makeStyles-secondary-7"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiFab-label"
  >
    <i
      className="natds-icons natds-icons-outlined-action-add makeStyles-root-8"
    />
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;
